### PR TITLE
privacy policy and TOS

### PR DESCRIPTION
**Pull request by ttoomey**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch

### Description
Adds a compliance agreement feature with admin-managed content pages that block access until users accept required policies.

- **Admin Content Management**: Adds CRUD for content pages, linking them to compliance requirements and warning when linked to active policies.
- **Compliance Enforcement**: Adds compliance requirements with activation/deactivation and agreement capture flow
- **Routing & Navigation**: Adds public content page viewing and compliance agreement routes plus admin menu entries
- **Test Coverage & Docs**: Adds factories, model specs, request specs for admin/public flows, and feature documentation for compliance requirements.

### Type of Change
New feature

### Type of Change
New feature
## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] I have updated the documentation (or not applicable)
- [x] I have added spec tests (or not applicable)
- [x] I have provided testing instructions in this PR or the related issue (or not applicable)

### CODE REVIEW

**Comment by ttoomey** [comment]
[2 tools called]

# QA Instructions

### What Changed
A new system allows admins to create mandatory "Compliance Requirements" (like Terms of Service) that users must agree to before accessing the application.

### Testing Focus
**Primary Testing** (Must verify):
- **Admin Setup**: Creating a Content Page and linking it to a new Compliance Requirement.
- **Enforcement**: Users are blocked from the app until they agree to active requirements.
- **Re-agreement**: Bumping the "Revision" number forces users to agree again.
- **Public Access**: Content pages are viewable without logging in (e.g., for footer links).

**Regression Check** (Quick verification):
- Login flow works normally when no requirements are active.
- Admin menu loads correctly with new "Content Pages" and "Compliance Requirements" options.

### Test Steps

**1. Setup a New Requirement (Admin)**
1. Log in as an Admin.
2. Navigate to **Admin** > **Content Pages**.
3. Click **New Content Page**.
   - Title: "QA Terms"
   - Slug: `qa_terms` (or auto-generated)
   - Content: Add some text.
   - Click **Save**.
4. Navigate to **Admin** > **Compliance Requirements**.
5. Click **New Compliance Requirement**.
   - Name: "Terms of Service"
   - Content Page: Select "QA Terms".
   - Revision: `1`
   - Click **Save**.

**2. Verify User Enforcement**
1. Log out or open an Incognito window.
2. Log in as any user.
3. **Verify** you are immediately redirected to the "Terms of Service" agreement page.
4. Try to click "Continue" without checking the box. **Verify** it is disabled or blocks you.
5. Check "I agree" and click "Continue".
6. **Verify** you are redirected to the dashboard/home page.

**3. Test Re-agreement (Version Bump)**
1. Log back in as Admin.
2. Go to **Admin** > **Compliance Requirements**.
3. Edit the "Terms of Service" requirement.
4. Change **Revision** from `1` to `2`.
5. Click **Save**.
6. Navigate to the dashboard (or any other page).
7. **Verify** you are immediately redirected to the agreement page again.
8. Agree and continue.

**4. Test Public Access**
1. Log out.
2. Navigate directly to `/pages/qa_terms` (replace `qa_terms` with the slug you used).
3. **Verify** the page content loads without requiring login.

**Note**: Unit tests cover expiration dates, multiple pending requirements, and edge cases where agreements expire. QA focus should be on

 the main setup and enforcement flow.

File: config/routes.rb [hunk 1]
```diff
@@ -38,6 +38,8 @@ def matches?(request)
   end
 
   get '/user_training', to: 'user_training#index'
+  resources :content_pages, only: [:show], param: :slug, path: 'pages'
```
Comment (eanders) [inline config/routes.rb]: If these are truly public and don't require authentication, they should probably be located under `/public/`.  This isn't really a problem now, but it makes it clear that no authentication is required, and when we get to oauth2-proxy, that path should already be cleared for access.  `/public/pages/slug` would be fine with me.

Comment (eanders) [reply config/routes.rb]: If these require authentication, but not authorization, they are good here.

**Review by eanders — APPROVED** [review]
@ttoomey I think this is good, just the one change, if those content pages should be available without authentication, we should move them.  Thanks!

File: Gemfile.lock [hunk 2]
```diff
@@ -1127,6 +1127,7 @@ GEM
 PLATFORMS
   arm64-darwin-22
   arm64-darwin-24
+  ruby
```
Comment (ttoomey) [inline Gemfile.lock:1130]: adding this again as bundler keeps insisting on it

File: config/routes.rb (hunk 1 above)
Comment (ttoomey) [reply config/routes.rb]: they are public, added namespace as suggested

File: app/views/admin/content_pages/_form.haml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+.row
+  .col-sm-8
+    - url = @page.persisted? ? admin_content_page_path(@page) : admin_content_pages_path
+    = simple_form_for @page, url: url, as: :content_page do |f|
+      = f.error_notification
+      .form-inputs
+        - slug_hint = 'URL-friendly identifier. Page will be available at (https://my.site/public/pages/<slug>). Cannot be changed after creation.'
```
Comment (eanders) [inline app/views/admin/content_pages/_form.haml:7]: do you think it makes sense to actually use the hostname here instead of `my.site`.  Should be available in `ENV['FQDN']`

**Review by eanders — APPROVED** [review]

File: app/views/admin/content_pages/_form.haml (hunk 3 above)
Comment (ttoomey) [reply app/views/admin/content_pages/_form.haml:7]: Good thought, that's nice QOL. I'm going to merge this into 192 as-is and address along with any other issues found by QA
